src/sensors/mpu9250sensor.cpp

```diff
@@ -166,16 +182,29 @@ void MPU9250Sensor::motionLoop() {
 
     quaternion = correction * quat;
 #else
-    unsigned long now = micros();
-    unsigned long deltat = now - last; //seconds since last update
-    last = now;
-    getMPUScaled();
+
+    union fifo_sample_raw buf;
+    uint16_t remaining_samples;
+    // TODO: would it be faster to read multiple samples at once
+    while (getNextSample (&buf, &remaining_samples)) {
+        parseAccelData(buf.sample.accel);
+        parseGyroData(buf.sample.gyro);
+        parseMagData(buf.sample.mag);
+
+        // TODO: monitor magnetometer status
+        // buf.sample.mag_status;
+        // TODO: monitor interrupts
+        // imu.getIntStatus();
+        // TODO: monitor remaining_samples to ensure that the number is going down, not up.
+        // remaining_samples
```

Comment:
* actually looking at the magnetometer status would allow us to detect when the magnetometer reading is incorrect
 * looking at the interrupt status would allow us to handle/detect fifo overflow
 * monitoring remaining_samples would allow is to avoid an infinite loop scenario in the case that our sample processing loop is *slower* than the imu outputs data.